proto/buf/alpha/registry/v1alpha1/request_builder.proto

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "buf/alpha/image/v1/image.proto";
+
+// ModuleMethod represents method definitions declared in a module's services.
+message ModuleMethod {
+  // The name of the method, e.g. "Foo"
+  string name = 1;
+
+  // A fully qualified name of the request type for the method, e.g.
+  // "pkg.baz.FooRequest"
+  string input_type = 2;
+  // A fully qualified name of the request type for the method, e.g.
+  // "pkg.baz.FooResponse"
+  string output_type = 3;
+
+  // Indicates if the method is annotated as deprecated.
+  bool deprecated = 4;
+  // Indicates if the method is client streaming.
+  bool client_streaming = 5;
+  // Indicates if the method is server streaming.
+  bool server_streaming = 6;
+}
+
+// ModuleService represents service definitions declared in a module.
+message ModuleService {
+  // The fully qualified name of the service, e.g. "pkg.foo.BarService"
+  string name = 1;
+
+  // The methods available in the service.
+  repeated ModuleMethod methods = 2;
+
+  // Indicates if the service is marked as deprecated.
+  bool deprecated = 3;
+}
+
+// RequestBuilderService serves endpoints used by requestbuilder.
+service RequestBuilderService {
+  // GetServices returns an overview of all services declared in a module
+  // and their methods.
+  rpc GetServices(GetServicesRequest) returns (GetServicesResponse);
+
+  // GetMethodDetails returns data needed to make dynamic requests for a method.
+  rpc GetMethodDetails(GetMethodDetailsRequest) returns (GetMethodDetailsResponse);
+}
```

Comment:
It looks to me like this service would provide all necessary information to allow users to quickly browse trough a list of services and methods (or search by name), and show the most relevant information (name, method type, deprecation) at the same time. This should be quite efficient too, and allow for sharing of a request as well.

Just a thought: If it weren't for the jsonschema, I _think_ we could get away with just `rpc GetServices`, and simply get a (partial) image for the service type. Obviously there would be a bit of overhead of retrieving all methods of the service where we only need one, but it might be a worthy trade-off. We do however have jsonschema, and I'm glad we do :sweat_smile:.